nomad/plan_apply.go

```diff
@@ -232,6 +261,39 @@ func (p *planner) applyPlan(plan *structs.Plan, result *structs.PlanResult, snap
 	return future, nil
 }
 
```

Comment:
Pls add comments to all these internal functions for future readability. Comments should make it very clear that the fields set are the only ones the plan applier is responsible for setting and everything else is looked up